### PR TITLE
Fix issue #27055

### PR DESCRIPTION
**Pull request by VALLIS-NERIA**
### Bugs this fixes
https://github.com/dotnet/roslyn/issues/27055

### CODE REVIEW

File: src/Compilers/Core/Portable/Syntax/SyntaxNode.cs [hunk 1]
```diff
@@ -433,7 +433,7 @@ public bool ContainsDiagnostics
         /// </summary>
         public bool Contains(SyntaxNode node)
         {
-            if (node == null || !this.FullSpan.Contains(node.FullSpan))
+            if (node == null || !this.FullSpan.Contains(node.FullSpan) || node == this)
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Syntax/SyntaxNode.cs]: > || node == this [](start = 70, length = 16)

It looks like this change broke a lot of existing tests, CI jobs are failing. Also, it looks like @gafter suggested to Won't Fix the issue and update API documentation instead.

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 1)

**Comment by VALLIS-NERIA** [comment]
Hmm, seems this behavior is not a bug but a feature.

**Comment by jcouv** [comment]
@VALLIS-NERIA  It sounds like this PR should be closed (issue is Won't-Fix).

**Pull request closed without merging**